core/mm/vm.c

```diff
@@ -1159,11 +1159,16 @@ static TEE_Result tee_mmu_user_va2pa_attr(const struct user_mode_ctx *uctx,
 			offset = region->offset +
 				 ROUNDDOWN((vaddr_t)ua - region->va, granule);
 
-			res = mobj_get_pa(region->mobj, offset, granule, &p);
-			if (res != TEE_SUCCESS)
-				return res;
-
-			*pa = p | ((vaddr_t)ua & (granule - 1));
+			if (mobj_is_paged(region->mobj)) {
+				*pa = 0;
```

Comment:
I'd prefer if we return an error code here. Perhaps `TEE_ERROR_NOT_SUPPORTED` to clearly set it apart from `TEE_ERROR_ACCESS_DENIED`. We could update `mobj_with_fobj_get_pa()` to return `TEE_ERROR_NOT_SUPPORTED` instead of `TEE_ERROR_GENERIC()` when `f->fobj->ops->get_pa` is `NULL`.

Reply:
ok, i'll look at your suggestion.